contracts/strategies/StrategyAaveLendingPoolV2.sol

```diff
@@ -0,0 +1,172 @@
+
+    // Info of supplying erc20 token to Aave lending pool
+    // The symbol of the supplying token
+    string public symbol;
+    // The address of supplying token (e.g. DAI, USDT)
+    address public supplyToken;
+
+    // The address of Aave interest-bearing token (e.g. aDAI, aUSDT)
+    address public aToken;
+
+    address public controller;
+
+    // The address of Aave Incentives Controller
+    address public incentivesController;
+
+    // The address of Aave StakeToken contract
+    address public stakeToken;
+
+    // The address of Aave token
+    address public aave;
+
+    address public uniswap;
+    // The address of WETH token
+    address public weth;
+
+    constructor(
+        address _lendingPool,
+        string memory _symbol,
+        address _supplyToken,
+        address _aToken,
+        address _controller,
+        address _incentivesController,
+        address _stakeToken,
+        address _aave,
+        address _uniswap,
+        address _weth
+    ) {
+        lendingPool = _lendingPool;
+        symbol = _symbol;
+        supplyToken = _supplyToken;
+        aToken = _aToken;
+        controller = _controller;
+        incentivesController = _incentivesController;
+        stakeToken = _stakeToken;
+        aave = _aave;
+        uniswap = _uniswap;
+        weth = _weth;
+    }
+
+    function getAssetAddress() external view override returns (address) {
+        return supplyToken;
+    }
+
+    function syncBalance() external view override returns (uint256) {
+        // Supplying token(e.g. DAI, USDT) balance of this contract.
+        // aToken value is pegged to the value of supplying erc20 token at a 1:1 ratio.
+        uint256 supplyTokenBalance = IAToken(aToken).balanceOf(address(this));
+        return supplyTokenBalance;
+    }
+
+    function harvest() external override {
+
+        // 1. Claims the StkAAVE staking rewards
+        uint256 stakingRewards = IStakeAave(stakeToken).getTotalRewardsBalance(address(this));
+
+        if (stakingRewards > 0) {
+            IStakeAave(stakeToken).claimRewards(address(this), stakingRewards);
+        }
+
+        // 2. Redeems staked tokens, note that it won't be success while this account not in redeem-available period 
+        stakeToken.call(abi.encodeWithSignature("redeem(address,uint256)", address(this), IERC20(stakeToken).balanceOf(address(this))));
+
+        // 3. Claims the liquidity incentives
+        address[] memory assets = new address[](1);
+        assets[0] = aToken;
+        uint256 rewardsBalance = IAaveIncentivesController(incentivesController).getRewardsBalance(assets, address(this)); 
+
+        if (rewardsBalance > 0) {
+            IAaveIncentivesController(incentivesController).claimRewards(assets, rewardsBalance, address(this));
+        }
+
+        // 4. Activates the cooldown period if there is balance in the aave safety module now 
+        if (IERC20(stakeToken).balanceOf(address(this)) > 0) {
```

Comment:
Are we able to check if we are resetting the cooldown here?

Reply:
As we have redeemed the staked token in step 2, and stakes new in step 3, so we can safely initiate a new cooldown here.

Reply:
Ah, right.

Reply:
We might want to also enforce a minimum harvest amount so that we don't waste cooldown for a very small reward.